app/models/miq_group.rb

```diff
@@ -64,7 +64,7 @@ def self.seed
     roles  = MiqUserRole.where("name like 'EvmRole-%'").index_by(&:name)
 
     role_map.each_with_index do |(group_name, role_name), index|
-      group = groups[group_name] || new(:description => group_name)
+      group = MiqGroup.where(description: group_name, tenant_id: Tenant.root_tenant).first || new(:description => group_name)
```

Comment:
The introduction of `groups` was to avoid the N+1. This adds it back.

Introduce this change 3 lines up creating `groups`.

```ruby
groups = where(:group_type => SYSTEM_GROUP, tenant_id: Tenant.root_tenant).includes(:entitlement).index_by(&:description)
```